src/convert.rs

```diff
@@ -814,12 +814,39 @@ pub fn elf_to_tbf(
             }
         }
 
-        // Save the end of this segment so we can check if padding is required
-        // between segments.
-        last_segment_address_end = Some(end_segment as usize);
+        if reloc_only {
+            if verbose {
+                println!("    Skipping segment since it only contains relocs");
+            }
+        } else {
+            // Handle padding now that we know we want something from this
```

Comment:
Can we just move the entire padding check from above here? I'd like to keep the utility of the option of `last_segment_address_end` and not have `padding == 0` mean both no padding and there can't be padding because this is the first segment.